src/routes/Trip/Items.tsx

```diff
@@ -64,9 +75,21 @@ const Items = ({
   if (isListLoading) {
     return <p className="mt-32  font-sans text-lg font-bold text-center  ">loading...</p>;
   }
-
   if (isListPresent) {
-    const itemsByState = listsById[currentTrip.listId].itemIds.reduce(
+    //ITEM FILTER
+    if (itemQuery === '') {
+      //Check if form field has any value
+      listOfItemIds = listsById[currentTrip.listId].itemIds;
+    } else {
+      const itemFilter = listsById[currentTrip.listId].items.filter((item: any) =>
+        item.label.toLowerCase().includes(itemQuery.toLowerCase()),
+      );
+      itemFilter.forEach((item: any) => {
+        return itemFilterIds.push(item.id);
+      });
+      listOfItemIds = itemFilterIds; //New List of items product of filtering
```

Comment:
As discussed in Slack and suggested above, we can filter the `itemIds` directly, using the `items` object to perform the comparison.

This way we save ourselves the extra loop to push the ids, and also fix the bug! 

```
      filteredItemIds = listsById[currentTrip.listId].itemIds.filter((itemId:any)=>{
        return items[itemId].label.toLowerCase().includes(itemQuery.toLocaleLowerCase());
      });
```